solidity/test/MezoAllocator.test.ts

```diff
@@ -542,32 +733,142 @@ describe("MezoAllocator", () => {
+          })
+        })
+      })
+
       context("when there is a deposit", () => {
-        let tx: ContractTransactionResponse
+        beforeAfterSnapshotWrapper()
+
+        const depositAmount = to1e18(5)
 
         before(async () => {
-          await tbtc.mint(await stbtc.getAddress(), to1e18(5))
+          await tbtc.mint(await stbtc.getAddress(), depositAmount)
           await mezoAllocator.connect(maintainer).allocate()
-          tx = await mezoAllocator.connect(governance).releaseDeposit()
         })
 
-        it("should emit DepositReleased event", async () => {
-          await expect(tx)
-            .to.emit(mezoAllocator, "DepositReleased")
-            .withArgs(1, to1e18(5))
-        })
+        context("when there is no donation", () => {
+          beforeAfterSnapshotWrapper()
+
+          let tx: ContractTransactionResponse
+
+          before(async () => {
+            tx = await mezoAllocator.connect(governance).releaseDeposit()
+          })
+
+          it("should emit DepositReleased event", async () => {
+            await expect(tx)
+              .to.emit(mezoAllocator, "DepositReleased")
+              .withArgs(1, depositAmount)
+          })
+
+          it("should decrease tracked deposit balance amount to zero", async () => {
+            const depositBalance = await mezoAllocator.depositBalance()
+            expect(depositBalance).to.equal(0)
+          })
+
+          it("should transfer tBTC to stBTC contract", async () => {
+            await expect(tx).to.changeTokenBalances(
+              tbtc,
+              [mezoPortal, stbtc],
+              [-depositAmount, depositAmount],
+            )
+          })
 
-        it("should decrease tracked deposit balance amount to zero", async () => {
-          const depositBalance = await mezoAllocator.depositBalance()
-          expect(depositBalance).to.equal(0)
+          it("should call MezoPortal.withdraw function", async () => {
+            await expect(tx)
+              .to.emit(mezoPortal, "WithdrawFully")
+              .withArgs(await tbtc.getAddress(), 1)
+          })
         })
 
-        it("should decrease Mezo Portal balance", async () => {
-          await expect(tx).to.changeTokenBalances(
-            tbtc,
-            [mezoPortal, stbtc],
-            [-to1e18(5), to1e18(5)],
-          )
+        context("when there is a donation", () => {
+          beforeAfterSnapshotWrapper()
+
+          const donationAmount = to1e18(2)
+
+          let tx: ContractTransactionResponse
+
+          before(async () => {
+            await tbtc.mint(await mezoAllocator.getAddress(), donationAmount)
+
+            tx = await mezoAllocator.connect(governance).releaseDeposit()
+          })
+
+          it("should emit DepositReleased event", async () => {
+            await expect(tx)
+              .to.emit(mezoAllocator, "DepositReleased")
+              .withArgs(1, depositAmount)
```

Comment:
Can we extract `1` to variable`const depositId = 1`? At first sight, it's hard to understand what `1` means (I had to check the `DepositReleased` event).

Reply:
93a781f0aced7c7a5f2a7c7e0cb3be0a0bf47077